### ISSUE TITLE
random context canceled error while reading the response body

### ISSUE THREAD

**Issue by ratanphayade** [gavv/httpexpect#119]
Hello, I'm using this package for e2e test runs on staging environment and I'm randomly getting `context canceled` error when I try running the below code.

```go
client := httpexpect.New(T(), "https://service.stage.com").
		GET("/status").
		WithTimeout(30 * time.Second).
		Expect()
```

Sample result:
```
reporter.go:23: 
        	Error Trace:	reporter.go:23
        	            		chain.go:21
        	            		response.go:107
        	            		response.go:84
        	            		request.go:1201
        	            		request.go:1155
        	            		identifier_test.go:133
        	Error:      	context canceled
```

Looking further into it, the error originated in the below line: https://github.com/gavv/httpexpect/blob/3894d81aa3bea64af6946448d76c3b0afe8b6df7/response.go#L105

So while digging deeper found that 
https://github.com/gavv/httpexpect/blob/3894d81aa3bea64af6946448d76c3b0afe8b6df7/request.go#L1339
is causing the issue. `defer cancel()` is called before `r.Body.Close()` is called which seems to be the issue. 

Need help here as this issue is surfacing randomly and could not reproduce it with defined steps. Please suggest if I'm missing something here.

Thanks,

**Comment by ratanphayade** [comment]
I did not observe any failures when commenting below line https://github.com/gavv/httpexpect/blob/3894d81aa3bea64af6946448d76c3b0afe8b6df7/request.go#L1339

**Comment by gavv** [comment]
Hi, thanks for detailed report!

I was able to reliably reproduce the issue and after digging into it, found out that there are several related bugs:

* when response has large body, and timeout is enabled, cancel is called before reading the body
* request and response bodies are not explicitly closed after reading
* if printer reads request or response body, it causes problems

I was able to create tests for all these cases and implemented a fix based on your idea from #118. Pushed to master. Thanks!
